plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergParquetFileWriter.java

```diff
@@ -106,19 +112,46 @@ public long getMemoryUsage()
     @Override
     public void appendRows(Page dataPage)
     {
-        parquetFileWriter.appendRows(dataPage);
+        try {
+            parquetFileWriter.appendRows(dataPage);
+        }
+        catch (TrinoException e) {
+            if (e.getErrorCode() == HIVE_WRITER_DATA_ERROR.toErrorCode()) {
```

Comment:
having Iceberg and Delta connectors remap error codes in a bunch of places, i would rather think we should

- consider this OK for Iceberg and Delta to reuse hive error codes (no remapping)
- decouple parquet writer from hive, so that it doesn't use hive error codes (eg if we provide connector specific error codes in the constructor, no need for remapping)

cc @raunaqmorarka @electrum

Reply:
I see. Anyway this fix doesn't seem to make sense.